### PR TITLE
Examples for `skip` and `stop`

### PR DESCRIPTION
**Pull request by mcabbott**
`skip` claims that if used in the callback passed to `train!`, the `update!` step will not happen. This isn't true, it's called too late for that. Its tests don't try to test that, instead they test that if used in the loss function, it doesn't evaluate the rest of the loss.

So right now this PR adds an example showing what it does do. Although perhaps better to simply remove it? Github's search can't find a single use of it in the wild. Originally added in #1232.

Also adds an example for `stop`, which does work as described.

### CODE REVIEW

**Comment by ToucheSir** [comment]
I have definitely seen it on code posted to Discourse, so removal might be too disruptive. The "proper" design would be to allow users to specify where in the training loop each callback runs, but it's a slippery slope from there to something like FluxTraining.jl.
